src/main/scala/cromwell/Main.scala

```diff
@@ -56,48 +56,55 @@ object Main extends App {
     * @param args The command line arguments.
     */
   private def setupServerLogging(args: Array[String]): Unit = {
-    args.headOption.map(_.capitalize) match {
-      case Some("Server") => sys.props.getOrElseUpdate("LOG_MODE", "STANDARD")
+    getAction(args) match {
+      case Some(Actions.Server) => sys.props.getOrElseUpdate("LOG_MODE", "STANDARD")
       case _ =>
     }
   }
+
+  private def getAction(args: Seq[String]): Option[Actions.Value] = for {
+    arg <- args.headOption
+    argCapitalized = arg.capitalize
+    action <- Actions.values find (_.toString == argCapitalized)
+  } yield action
 }
 
 /** A simplified version of the Akka `PromiseActorRef` that doesn't time out. */
-private class PromiseWorkflowActor(promise: Promise[Any], runner: ActorRef) extends Actor {
-  runner ! RunWorkflow
-
+private class PromiseActor(promise: Promise[Any]) extends Actor {
   override def receive = {
-    case Status.Failure(f) => promise.tryFailure(f)
-    case success => promise.trySuccess(success)
+    case Status.Failure(f) =>
+      promise.tryFailure(f)
+      context.stop(self)
+    case success =>
+      promise.trySuccess(success)
+      context.stop(self)
   }
 }
 
-
-class Main private[cromwell](enableTermination: Boolean, managerSystem: () => WorkflowManagerSystem) {
+class Main private[cromwell](managerSystem: WorkflowManagerSystem) {
   private[cromwell] val initLoggingReturnCode = initLogging()
 
   lazy val Log = LoggerFactory.getLogger("cromwell")
   Monitor.start()
 
-  def this() = this(enableTermination = true, managerSystem = () => new WorkflowManagerSystem {})
+  def this() = this(managerSystem = new WorkflowManagerSystem {})
 
   // CromwellServer still doesn't clean up... so => Any
-  def runAction(args: Seq[String]): Any = {
-    getAction(args.headOption) match {
+  def runAction(args: Seq[String]): Int = {
+    Main.getAction(args) match {
       case Some(x) if x == Actions.Validate => validate(args.tail)
       case Some(x) if x == Actions.Highlight => highlight(args.tail)
       case Some(x) if x == Actions.Inputs => inputs(args.tail)
       case Some(x) if x == Actions.Run => run(args.tail)
       case Some(x) if x == Actions.Parse => parse(args.tail)
-      case Some(x) if x == Actions.Server => CromwellServer
+      case Some(x) if x == Actions.Server => CromwellServer.run()
```

Comment:
Also not new with your changes, but nothing complains if there are extra args to `server`.